bindings/python/FloatingBaseEstimators.cpp

```diff
@@ -0,0 +1,147 @@
+        .def_readwrite("ekf_update_enabled", &Options::ekfUpdateEnabled)
+        .def_readwrite("acceleration_due_to_gravity", &Options::accelerationDueToGravity);
+
+    py::class_<SensorsStdDev>(module, "FloatingBaseEstimatorSensorsStdDev")
+        .def(py::init())
+        .def_readwrite("accelerometer_noise", &SensorsStdDev::accelerometerNoise)
+        .def_readwrite("gyroscope_noise", &SensorsStdDev::gyroscopeNoise)
+        .def_readwrite("accelerometer_bias_noise", &SensorsStdDev::accelerometerBiasNoise)
+        .def_readwrite("gyroscope_bias_noise", &SensorsStdDev::gyroscopeBiasNoise)
+        .def_readwrite("contact_foot_lin_vel_noise", &SensorsStdDev::contactFootLinvelNoise)
+        .def_readwrite("contact_foot_ang_vel_noise", &SensorsStdDev::contactFootAngvelNoise)
+        .def_readwrite("swing_foot_lin_vel_noise", &SensorsStdDev::swingFootLinvelNoise)
+        .def_readwrite("swing_foot_ang_vel_noise", &SensorsStdDev::swingFootAngvelNoise)
+        .def_readwrite("forward_kinematics_noise", &SensorsStdDev::forwardKinematicsNoise)
+        .def_readwrite("encoders_noise", &SensorsStdDev::encodersNoise);
+
+    py::class_<Advanceable<Output>>(module, "FloatingBaseEstimatorOutputAdvanceable");
+
+    py::class_<FloatingBaseEstimator, Advanceable<Output>> floatingBaseEstimator(module, "FloatingBaseEstimator");
+
+    floatingBaseEstimator.def(py::init())
+        .def("set_imu_measurement", &FloatingBaseEstimator::setIMUMeasurement,
+             py::arg("acc_meas"), py::arg("gyro_meas"))
+        .def("set_contact_status", &FloatingBaseEstimator::setContactStatus,
+             py::arg("name"), py::arg("contact_status"), py::arg("switch_time"), py::arg("time_now"))
+        .def("set_kinematics", &FloatingBaseEstimator::setKinematics,
+             py::arg("encoders"), py::arg("encoder_speeds"))
+        .def("advance", &FloatingBaseEstimator::advance)
+        .def("reset_estimator", py::overload_cast<const InternalState&>(&FloatingBaseEstimator::resetEstimator),
+             py::arg("new_state"))
+        .def("reset_estimator", py::overload_cast<const Eigen::Quaterniond&, const Eigen::Vector3d&>(&FloatingBaseEstimator::resetEstimator),
+             py::arg("new_base_orientation"), py::arg("new_base_position"))
+        .def("get", &FloatingBaseEstimator::get)
+        .def("is_valid", &FloatingBaseEstimator::isValid)
+        .def("initialize",
+            [](FloatingBaseEstimator& impl, std::shared_ptr<IParametersHandler> handler, std::shared_ptr<iDynTree::KinDynComputations> kinDyn) -> bool {
+                return impl.initialize(handler, kinDyn);
+            },
+            py::arg("handler"), py::arg("kindyn"));
+
+    // at the moment at an interface level, this nested class need not be exposed
+    // but in case, we want to use the FloatingBaseEstimator as base class
+    // for python based implementations, maybe it could be useful
```

Comment:
This is definitely possible with pybind, but it requires quite a lot of work since it demands creating a dummy class (called trampoline) for the interface. In my opinion, once the entire infrastructure is up and running, this could be a great alternative for prototyping, but as you wrote, let's keep this option for future work.